webapp/src/ts/components/actionbar/actionbar.component.html

```diff
@@ -148,79 +169,102 @@
+          <a *ngIf="actionBar?.right?.sendTo"
+            mmAuth="can_view_call_action"
+            class="mm-icon mm-icon-inverse mm-icon-caption mobile-only"
+            [attr.href]="'tel:' + actionBar?.right?.sendTo?.phone | safeHtml:'url'"
+            [ngClass]="{ 'mm-icon-disabled': !actionBar?.right?.sendTo?.phone }"
+          >
             <span class="fa fa-phone"></span>
             <p translate>call</p>
           </a>
-          <a class="mm-icon mm-icon-inverse mm-icon-caption mobile-only" href="sms:{{actionBar?.right?.sendTo?.phone}}" [ngClass]="{'mm-icon-disabled': !actionBar?.right?.sendTo?.phone}" *ngIf="actionBar?.right?.sendTo" mmAuth="can_view_message_action">
+          <a *ngIf="actionBar?.right?.sendTo"
+            mmAuth="can_view_message_action"
+            class="mm-icon mm-icon-inverse mm-icon-caption mobile-only"
+            [attr.href]="'sms:' + actionBar?.right?.sendTo?.phone | safeHtml:'url'"
+            [ngClass]="{ 'mm-icon-disabled': !actionBar?.right?.sendTo?.phone }"
+          >
             <span class="fa fa-envelope"></span>
             <p translate>Send Message</p>
           </a>
-          <a class="mm-icon mm-icon-inverse mm-icon-caption send-message desktop-only" data-send-to="{{actionBar?.right?.sendTo?._id}}" [ngClass]="{'mm-icon-disabled': !actionBar?.right?.sendTo?.phone}" *ngIf="actionBar?.right?.sendTo" mmAuth="can_view_message_action">
+          <a *ngIf="actionBar?.right?.sendTo"
+            mmAuth="can_view_message_action"
+            class="mm-icon mm-icon-inverse mm-icon-caption send-message desktop-only"
+            [attr.data-send-to]="actionBar?.right?.sendTo?._id"
+            [ngClass]="{ 'mm-icon-disabled': !actionBar?.right?.sendTo?.phone }"
+            (click)="actionBar?.right?.sendTo?.phone && actionBar?.right?.openSendMessageModal(actionBar?.right?.sendTo?._id)"
+          >
             <span class="fa fa-envelope"></span>
             <p translate>Send Message</p>
           </a>
 
-          <a class="mm-icon mm-icon-inverse mm-icon-caption" ui-sref="contacts.edit({ id: selectedContactDoc._id })" [ngClass]="{'mm-icon-disabled': !actionBar?.right?.canEdit}">
+          <a class="mm-icon mm-icon-inverse mm-icon-caption"
+            [routerLink]="['/', 'contacts', selectedContactDoc._id, 'edit']"
+            [ngClass]="{'mm-icon-disabled': !actionBar?.right?.canEdit}"
+          >
             <span class="fa fa-pencil"></span>
             <p translate>Edit</p>
           </a>
 
-          <a class="mm-icon mm-icon-inverse mm-icon-caption" (click)="deleteDoc(selectedContactDoc)" [ngClass]="{'mm-icon-disabled': !actionBar.right.canDelete}" mmAuth="can_delete_contacts">
+          <a mmAuth="can_delete_contacts"
+            class="mm-icon mm-icon-inverse mm-icon-caption"
+            (click)="deleteDoc(selectedContactDoc.doc)"
```

Comment:
So this is where the problem with your delete came from? `selectedContactDoc` is not `selectedContact.doc` anymore? 
I think the selector the actionbar is using is different then. In the old code, we had a selector for `selectedContact.doc` here: https://github.com/medic/cht-core/blob/master/webapp/src/js/selectors/index.js#L38

Is it not simpler and more efficient to use it instead of the whole `selectedContact`?

Reply:
I've switched to selectedContactDoc selector, so this should be receiving the selectedContactDoc now. Thank you!